### DISCUSSION TITLE
Error starting my server error ACE00046

### DISCUSSION THREAD

**Discussion by rm5hunter** [azerothcore/azerothcore-wotlk#10551] [General]
the error is that the version of mysql and the core are not compatible but it was working normally from one moment to another it stopped working

# Location '/home/acore/azerothcore/src/server/database/Database/DatabaseWorkerPool.cpp:76'
# Function 'DatabaseWorkerPool'
# Condition 'Used MySQL library version (%s id %lu) does not match the version id used to compile AzerothCore (id %u).
Search the wiki for ACE00046 in Common Errors (https://www.azerothcore.org/wiki/common-errors).'
# Message '8.0.28'

**Comment by Kitzunu** [comment]
Read the error code `ACE00046` in the link that the error message is providing you.

**Reply by rm5hunter** [reply]
I already went to that link but the solution appears for windows not for linux

**Reply by Kitzunu** [reply]
1. Pull the source.
2. Recompile.
3. Change your mysql libraries to the new ones as you have probably updated your mysql server

**Reply by rm5hunter** [reply]
I'm talking about a mysql.dll file and that the core and mysql must be in the same version but I don't know how to find the path of the dll to change and modify it I don't know the path I tried to look for in the folder etc mysql and etc bin but I did not find such a dll

**Reply by rm5hunter** [reply]
how I do this?

**Reply by Kitzunu** [reply]
dll is a windows file. Try to recompile your source

**Reply by rm5hunter** [reply]
i have in ubuntu 20.04

**Reply by rm5hunter** [reply]
**I have a question every time I get this error I must do what you tell me I see it very complex this error has already happened to me and I do not reinstall everything**

**Reply by Kitzunu** [reply]
The error happens when your **MySQL server version** is not matching the **MySQL Libraries that you compiled with** that you use.

So either you **updated your mysql server to a new version and recompiled but you did not update your library. So you update your library**
or **You have updated your libraries and mysql server but you have compiled binaries with an older library. So you just need to recompile the source**

I do not run linux so I am not 100% sure how it works. But these are the 2 reasons as to why you can get that error.

**Reply by rm5hunter** [reply]
but if the server has only been online for 10 days and this error seems very ugly to me, it is so easy to recompile

**Reply by rm5hunter** [reply]
Do you know of any manual to recompile in ubuntu linux

**Reply by Kitzunu** [reply]
as I said the error occurs when **your** mysql server version and mysql libraries does **not** match with the **version** you had when you compiled your worldserver and authserver binaries.

It is easily fixed by **recompiling** and updating the libraries (which I am not sure how as I do not use linux).

**Reply by Kitzunu** [reply]
Read the linux installation guide and you can learn how to recompile https://www.azerothcore.org/wiki/installation

**Reply by rm5hunter** [reply]
thank you but I am dissatisfied because it is a lot of work it is practically to reinstall everything

**Reply by Kitzunu** [reply]
That's how the cookie crumbles.
To avoid the servers breaking or to give unwanted results when querying to the database it is very important that the library you use is the **exact same version as the library you used to compile with**.

And recompiling is not reinstalling everything. Everytime you pull the source you have to recompile to get the changes. That is how C++ works.
